### PR TITLE
Turned exercise 021 into practice

### PR DESCRIPTION
**Pull request by divyankachaudhari**
Addresses #2198. Turns [Exercise 021](https://github.com/ocaml/ocaml.org/blob/main/data/exercises/021_insert_element.md) into practice.

Added four extra test cases:
- Insert at the beginning
- Insert at the end
- Insert beyond the list
- Insert into empty list

### CODE REVIEW

File: practice/021/ex.ml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        (Tested.insert_at "beta" 0 ["a"; "b"; "c"; "d"]));
+    "insert at the end" >:: (fun _ ->
+      assert_equal ["a"; "b"; "c"; "d"; "alfa"]
+        (Tested.insert_at "alfa" 4 ["a"; "b"; "c"; "d"]));
+    "insert beyond the end" >:: (fun _ ->
+      assert_equal ["a"; "b"; "c"; "d"; "beta"]
+        (Tested.insert_at "beta" 6 ["a"; "b"; "c"; "d"]));
+    "insert into empty list" >:: (fun _ ->
+      assert_equal ["alfa"]
+        (Tested.insert_at "alfa" 0 []));
```
Comment (cuihtlauac) [inline practice/021/ex.ml]:
```suggestion
      assert_equal ["a"; "alpha"; "b"; "c"; "d"]
        (Tested.insert_at "alpha" 1 ["a"; "b"; "c"; "d"]));
    "insert at the beginning" >:: (fun _ ->
      assert_equal ["beta"; "a"; "b"; "c"; "d"]
        (Tested.insert_at "beta" 0 ["a"; "b"; "c"; "d"]));
    "insert at the end" >:: (fun _ ->
      assert_equal ["a"; "b"; "c"; "d"; "alpha"]
        (Tested.insert_at "alpha" 4 ["a"; "b"; "c"; "d"]));
    "insert beyond the end" >:: (fun _ ->
      assert_equal ["a"; "b"; "c"; "d"; "beta"]
        (Tested.insert_at "beta" 6 ["a"; "b"; "c"; "d"]));
    "insert into empty list" >:: (fun _ ->
      assert_equal ["alpha"]
        (Tested.insert_at "alpha" 0 []));
```

**Review by cuihtlauac — CHANGES_REQUESTED** [review]

File: practice/021/ex.ml (hunk 1 above)
Comment (divyankachaudhari) [reply practice/021/ex.ml]: Had just followed the spelling from the markdown 😅

**Comment by divyankachaudhari** [comment]
@cuihtlauac Could you please review this PR? Thank you.

**Review by cuihtlauac — APPROVED** [review]

**Comment by cuihtlauac** [comment]
Thanks @divyankachaudhari

**Pull request merged**